WordPress/wordpress-playground
# Blueprints: Validate using JSON schema

At the moment, invalid Blueprints yield non-useful error messages such as "cannot read property X of undefined." This commit adds a JSON schema-based validation to the compileBlueprint() function to give developers better idea about what to fix.

The JSON schema used here is generated from Blueprint TypeScript type which means building the Blueprints package became a multi-step process:

1. Generate the type rollup
2. Generate the JSON schema
3. Bundle the package with the generated schema included

## Other considerations

* Should `packages/playground/blueprints/public/blueprint-schema.json` be included in the repo? It is a derived file so maybe it could be listed in `.gitignore` and just be regenerated every time `npm run dev` starts
* I documented how the JSON schema is not autogenerated in `npm run dev` mode. This is not ideal but I don't have a better solution right now. Any ideas how to make it better @dmsnell?

## Testing Instructions

1. Confirm the code makes sense
2. Confirm the local website still works, play with the query API options
3. Confirm the CI is green (this PR comes with tests)
